core/templates/dev/head/domain/objects/objects-domain.constants.ts

```diff
@@ -22,7 +22,8 @@ export class ObjectsDomainConstants {
       'Please only use numerical digits, spaces or forward slashes (/)',
     INVALID_FORMAT:
       'Please enter a valid fraction (e.g., 5/3 or 1 2/3)',
-    DIVISION_BY_ZERO: 'Please do not put 0 in the denominator'
+    DIVISION_BY_ZERO: 'Please do not put 0 in the denominator',
+    INTERMEDIATE: ''
```

Comment:
Looks like the message is empty.

Reply:
@kevinlee12 That was intentional. According to the issue #8126 there should be no error displayed when the user is still typing a negative expression

Reply:
Agreed on the user behavior, but the code is odd and doesn't look like the right way to do it as it's hacking the Error to throw another error.  Can we choose to suppress the error instead of throwing a new one?

Reply:
@kevinlee12 I'm sorry I didn't get what are you saying? If we choose not to throw an error, the intermediate expression will act as a valid expression i.e if a person types just `-` and clicks `OK` it will be treated as a valid expression and it will cause an error. Please correct me if I'm wrong.

Reply:
I think we're on the same page in terms of user behavior. I'm more concerned about how the code is implementing that behavior. We should do it in the likes of the `/` as well, which doesn't throw any errors. Looking at the code, I would suggest just not throwing any error when the format is okay instead of throwing an "empty" error. Does that sound ok?

Reply:
@kevinlee12 If we don't throw any error in an intermediate expression, it would treat it like a valid expression and that is wrong too.

Reply:
In that case, throwing an error that doesn't have a message will make the user confused as well then since they won't know what's wrong.

Reply:
There is no sign of error in frontend when we throw an empty error.

Reply:
Hmm, how will the user know if there's an issue if they do `12 - 3`?

Reply:
12 - 3 shows an error 

![Screenshot_20191208_004452](https://user-images.githubusercontent.com/36989112/70379547-fdd81c00-1953-11ea-9126-8846ccb70209.png)

Reply:
Thanks for confirming the UI, though I still don't think this code is the right solution (even if the UX is right). Is it possible to modify the logic below instead?

 https://github.com/oppia/oppia/blob/37285a9b0e7866cbcdf5f8193a6ac2fab64458a7/extensions/objects/templates/fraction-editor.directive.ts#L45-L50

Ultimately my main concern is that this is an anti-pattern so I feel like more work is necessary to not have it as an anti-pattern.

Reply:
Also would like @seanlip's thoughts on this as well.

Reply:
@kevinlee12 I have made a pr suggesting an alternative fix to this issue #8149 PTAL!